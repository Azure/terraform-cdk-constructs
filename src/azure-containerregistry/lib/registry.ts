import { ContainerRegistry } from "@cdktf/provider-azurerm/lib/container-registry";
import * as cdktf from "cdktf";
import { Construct } from "constructs";
import { AzureResource } from "../../core-azure/lib";

export interface RegistryProps {
  /**
   * The Azure Region to deploy.
   */
  readonly location: string;
  /**
   * The name of the Log Analytics Workspace.
   */
  readonly name: string;
  /**
   * The name of the Azure Resource Group.
   */
  readonly resource_group_name: string;
  /**
   * The SKU of the Log Analytics Workspace.
   */
  readonly sku?: string;
  /**
   * The tags to assign to the Resource Group.
   */
  readonly tags?: { [key: string]: string };
  /**
   * Create enable Admin user.
   */
  readonly admin_enabled?: boolean;
  /**
   * Specify the locations to configure replication.
   */
  readonly georeplication_locations?: any;
}

export class Registry extends AzureResource {
  public readonly props: RegistryProps;
  public readonly resourceGroupName: string;
  public readonly id: string;

  constructor(scope: Construct, id: string, props: RegistryProps) {
    super(scope, id);

    this.props = props;
    this.resourceGroupName = props.resource_group_name;

    // Provide default values
    const sku = props.sku ?? "Basic";
    const admin_enabled = props.admin_enabled ?? false;
    const georeplication_locations = props.georeplication_locations ?? [];

    const azurermContainerRegistry = new ContainerRegistry(this, "acr", {
      location: props.location,
      name: props.name,
      resourceGroupName: props.resource_group_name,
      sku: sku,
      tags: props.tags,
      adminEnabled: admin_enabled,
      georeplications: georeplication_locations,
    });

    this.id = azurermContainerRegistry.id;

    // Terraform Outputs
    const cdktfTerraformOutputACRid = new cdktf.TerraformOutput(this, "id", {
      value: azurermContainerRegistry.id,
    });

    const cdktfTerraformOutputACRName = new cdktf.TerraformOutput(
      this,
      "container_registry_name",
      {
        value: azurermContainerRegistry.name,
      },
    );

    const cdktfTerraformOutputACRloginserver = new cdktf.TerraformOutput(
      this,
      "login_server",
      {
        value: azurermContainerRegistry.loginServer,
      },
    );

    /*This allows the Terraform resource name to match the original name. You can remove the call if you don't need them to match.*/
    cdktfTerraformOutputACRid.overrideLogicalId("id");
    cdktfTerraformOutputACRName.overrideLogicalId("container_registry_name");
    cdktfTerraformOutputACRloginserver.overrideLogicalId("login_server");
  }
}
