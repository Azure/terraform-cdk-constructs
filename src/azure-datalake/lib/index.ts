export * from "./filesystem";
export * from "./path";
export * from "./datalake";
