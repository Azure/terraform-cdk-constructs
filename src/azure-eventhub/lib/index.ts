export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
