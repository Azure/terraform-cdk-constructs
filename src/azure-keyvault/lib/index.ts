export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

