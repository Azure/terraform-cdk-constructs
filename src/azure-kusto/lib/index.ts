export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
