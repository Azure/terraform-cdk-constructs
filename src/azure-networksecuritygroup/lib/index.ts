export * from './preconfigured-rules';
export * from './securitygroup';
