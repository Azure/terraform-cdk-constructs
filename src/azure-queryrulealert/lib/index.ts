export * from './query-rule-alert'