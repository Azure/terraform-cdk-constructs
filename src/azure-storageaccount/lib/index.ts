export * from './account';
export * from './container';
export * from './fileshare';
export * from './queue';
export * from './table';
