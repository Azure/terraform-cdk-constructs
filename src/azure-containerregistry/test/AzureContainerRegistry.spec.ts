import { Testing, TerraformStack} from 'cdktf';
import { exampleAzureContainerRegistry} from './ExampleAzureContainerRegistry'
import 'cdktf/lib/testing/adapters/jest';
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";
import { AzureContainerRegistry } from '../';

describe('Azure Container Registry With Defaults', () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {features: {}});
    new AzureContainerRegistry(stack, 'testAzureContainerRegistryDefaults', {
      name: "latest",
      location: 'eastus',
      resource_group_name: "rg-test",
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Container Registry with defaults and checks snapshot", () => {
    expect(
      Testing.synth(stack)
    ).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});


describe('Azure Container Registry Example', () => {
  
  it("renders the Azure Container Registry and checks snapshot", () => {
    expect(
      Testing.synth(new exampleAzureContainerRegistry(Testing.app(), "testAzureContainerRegistry"))
      
    ).toMatchSnapshot();
  });

  it("check if the produced terraform configuration is valid", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(new exampleAzureContainerRegistry(Testing.app(), "testAzureContainerRegistry"))).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
  
    
    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(new exampleAzureContainerRegistry(Testing.app(), "testAzureContainerRegistry"))).toPlanSuccessfully();
  });
});
