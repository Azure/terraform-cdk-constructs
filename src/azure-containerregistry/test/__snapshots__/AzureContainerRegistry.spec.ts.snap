// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Container Registry Example renders the Azure Container Registry and checks snapshot 1`] = `
"{
  "output": {
    "container_registry_name": {
      "value": "\${azurerm_container_registry.testACR_acr_53FDF08B.name}"
    },
    "id": {
      "value": "\${azurerm_container_registry.testACR_acr_53FDF08B.id}"
    },
    "login_server": {
      "value": "\${azurerm_container_registry.testACR_acr_53FDF08B.login_server}"
    },
    "resource_group_name": {
      "value": "\${azurerm_resource_group.rg.name}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_container_registry": {
      "testACR_acr_53FDF08B": {
        "admin_enabled": false,
        "georeplications": [
          {
            "location": "westus"
          }
        ],
        "location": "\${azurerm_resource_group.rg.location}",
        "name": "acr\${var.name}",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "sku": "Premium",
        "tags": {
          "environment": "test"
        }
      }
    },
    "azurerm_resource_group": {
      "rg": {
        "location": "eastus",
        "name": "rg-\${var.name}"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.75.0"
      }
    }
  },
  "variable": {
    "name": {
      "default": "test",
      "description": "System name used to randomize the resources",
      "type": "string"
    }
  }
}"
`;

exports[`Azure Container Registry With Defaults renders an Azure Container Registry with defaults and checks snapshot 1`] = `
"{
  "output": {
    "container_registry_name": {
      "value": "\${azurerm_container_registry.testAzureContainerRegistryDefaults_acr_93F9D5D6.name}"
    },
    "id": {
      "value": "\${azurerm_container_registry.testAzureContainerRegistryDefaults_acr_93F9D5D6.id}"
    },
    "login_server": {
      "value": "\${azurerm_container_registry.testAzureContainerRegistryDefaults_acr_93F9D5D6.login_server}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_container_registry": {
      "testAzureContainerRegistryDefaults_acr_93F9D5D6": {
        "admin_enabled": false,
        "georeplications": [
        ],
        "location": "eastus",
        "name": "latest",
        "resource_group_name": "rg-test",
        "sku": "Basic"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.75.0"
      }
    }
  }
}"
`;
